src/main/resources/db/changesets/20181203_studytable.yaml

```diff
@@ -0,0 +1,208 @@
+              - column:
+                  name: type
+                  type: varchar(128)
+
+        - createTable:
+            tableName: study_relationship
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: name
+                  type: varchar(256)
+              - column:
+                  name: from_cardinality
+                  type: varchar(32)
+              - column:
+                  name: to_cardinality
+                  type: varchar(32)
+              - column:
+                  name: from_column
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_study_relationship_study_column_from
+                    references: study_column(id)
+              - column:
+                  name: to_column
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_study_relationship_study_column_to
+                    references: study_column(id)
+
+        - createTable:
+            tableName: asset_specification
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: study_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_asset_specification_study
+                    references: study(id)
+              - column:
+                  name: name
+                  type: varchar(256)
+
+        - createTable:
+            tableName: asset_table
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: asset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_asset_table_asset_specification
+                    references: asset_specification(id)
+              - column:
+                  name: table_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_asset_table_study_table
+                    references: study_table(id)
+              - column:
```

Comment:
To support non-hierarchical asset specs, perhaps identify the root table in the asset_specification instead of here. Then we can (eventually) allow that to be NULL to indicate no root table.

Reply:
I like this. I removed the asset_table column as well, since this can be inferred from the column that the asset_column references.